src/index.js

```diff
@@ -1,14 +1,32 @@
-export const html = hypertext(string => {
+function renderHtml(string) {
   const template = document.createElement("template");
   template.innerHTML = string;
   return template.content;
-});
+}
 
-export const svg = hypertext(string => {
+function renderSvg(string) {
   const g = document.createElementNS("http://www.w3.org/2000/svg", "g");
   g.innerHTML = string;
   return g;
-});
+}
+
+export const html = Object.assign(hypertext(renderHtml, fragment => {
+  if (fragment.firstChild === null) return null;
+  if (fragment.firstChild === fragment.lastChild) return fragment.removeChild(fragment.firstChild);
+  const span = document.createElement("span");
+  span.appendChild(fragment);
+  return span;
+}), {fragment: hypertext(renderHtml, fragment => fragment)});
+
+export const svg = Object.assign(hypertext(renderSvg, g => {
+  if (g.firstChild === null) return null;
+  if (g.firstChild === g.lastChild) return g.removeChild(g.firstChild);
+  return g;
+}), {fragment: hypertext(renderSvg, g => {
+  const fragment = document.createDocumentFragment();
+  while (g.firstChild) fragment.appendChild(g.firstChild);
```

Comment:
I'm probably missing something obvious, but why is this a `while` loop?

Reply:
To capture all the children rather than just the first maybe?

Reply:
To capture all the children rather than just the first maybe?

Reply:
Because the G element can have multiple children. Appending the first child to the document fragment removes it from the G element, changing the value of g.firstChild, and the loop breaks when the G element is empty.